packages/extension-ui/src/partials/Header.tsx

```diff
@@ -45,22 +45,12 @@ function Header({
   withHelp,
   withSettings
 }: Props): React.ReactElement<Props> {
-  const [connectedActiveTabUrl, setConnectedActiveTabUrl] = useState<string | undefined>();
+  const connectedActiveTabUrl = useConnectedActiveTabUrl();
   const { t } = useTranslation();
 
-  const isConnected = !!connectedActiveTabUrl;
+  const isConnected = !!(withConnectedAccounts && connectedActiveTabUrl);
```

Comment:
not sure if `withConnectedAccounts` should be a part of this condition -- app can be connected even though there are no account, I guess :shrug:

Reply:
I was trying to replicate the previous logic, before changing to the new, extracted util. Don't you think it previously worked the same way?

Reply:
I think you are right, now I see it